### DISCUSSION TITLE
Custom Shader

### DISCUSSION THREAD

**Discussion by A9191** [sceneview/sceneform-android#90] [Q&A]
I would like to know how can I implement a custom shader to an image plane?
I think before it was possible to put .mat in .sfa, but how could I do it now?

**Comment by RGregat** [comment]
At the moment only predefined Materials can be used. I'm not sure how difficult it is to implement the support for custom Materials.

**Comment by ThomasGorisse** [comment]
You can have a look at how it's done in the [VideoNode](https://github.com/ThomasGorisse/sceneform-android-sdk/blob/81f7aa2d175b3c3240e59da1fb110bd4096a74c5/ux/src/main/java/com/google/ar/sceneform/ux/VideoNode.java)

You have 2 choices:
- Compile your .mat file to .filamat using the Filament matc tool : [Filament v1.9.25](https://github.com/google/filament/releases/tag/v1.9.25)
- Use the Filament  [filamat-android depency](https://github.com/google/filament#android) and write your .mat content directly in your java/kotlin code.

**Warning:** make sure to use the same Filament version as the current one used by Sceneform.
